Creado y modificado PUTS de producto

1862507J opened the pull request:
<!-- markdownlint-disable -->
## 🖼️ Contexto
<!-- markdownlint-enable -->

 <!-- Acá damos el contexto necesario para entender los cambios realizados.
 Piensa que esto lo va a leer alguien que no sabe nada
 respecto a estos cambios. -->

Se creó un endpoint especifico para solo modificar el stock (no permite modificación de ni un otro parámetro)
Se modificó el endpoint PUT normal, ya que contenía errores de DJANGO
Se crearon test para todos los casos

## 🛠️ Cambios realizados
<!-- > Acá va un punteo de los cambios que van incluidos en este PR, incluyendo
adiciones, sustracciones, reorganizaciones de archivos, etc. -->

Se creó un endpoint especifico para solo modificar el stock (no permite modificación de ni un otro parámetro)
Se modificó el endpoint PUT normal, ya que contenía errores de DJANGO
Se crearon test para todos los casos


## 🔚📍 Endpoints modificados
<!-- Opcional. Si los cambios incluyen la creación/modificación/eliminación de
endpoints, indícalos acá. -->


## ✅ Checklist
<!-- > Acá hay una lista de requisitos obligatorios que debes cumplir al momento de mergear tu PR. -->
- [x] Lo probé localmente.
- [x] Actualicé test antiguos.
- [x] Cree nuevos tests.
- [x] Actualicé la API interna en Postman.
- [x] Actualicé la API pública en Postman.
- [x] Revisé que las migraciones no conflictuen.

## 🧐 ¿Cómo se ve o cómo se prueba?
<!-- Opcional. Acá hacemos una mini demo de la nueva feature o bien explicamos
como probarla. Si son solo cambios de backend, puede servir un pantallazo de
postman llamando al endpoint afectado. -->

Con postman están los endpoints para probarlos a mano, existe un parámetro para el id para evitar tener que copiar y pegarlo en todos lados y se pueden correr los test

## ⏳ Pendientes

<!-- Otros puntos necesarios para completar la feature relacionada a los cambios
de este PR. Por ejemplo, cambios en otro proyecto, o alguna configuración de
deployment. También se pueden proponer mejoras futuras relacionadas a
estos cambios. -->
Falta implementar que solo un admin pueda hacerlo, y cambiar ciertos códigos para que no estén en el controlador

<!-- Usar la siguiente sintaxis: -->
<!-- - [ ] .. -->

1862507J committed:
Updated Gitignore

1862507J committed:
feat: Add discount_percentage, category, gender, size, color, and quantity fields to ProductSerializer

This commit adds the following fields to the `ProductSerializer`: `discount_percentage`, `category`, `gender`, `size`, `color`, and `quantity`. These fields are included in the `fields` list and are marked as read-only in the `read_only_fields` list. This update allows for more comprehensive serialization of the `Product` model.

Note: The commit message has been generated based on the provided code changes and recent repository commits.

1862507J committed:
created test folders

1862507J committed:
Updated product PUT

Había un error ya que no dejaba cambiar la descripción al ahcer put

1862507J committed:
feat: Created test for products puts url

This commit updates the `ProductViewSet` to allow changing the description when making a PUT request. Previously, the description field was not being updated even when provided in the request data. This change fixes the issue by checking if the description is present in the request data and updating it accordingly.

Note: The commit message has been generated based on the provided code changes and recent repository commits.

1862507J committed:
Test Agregado cuando stock update no tiene parametro stock

pedrodepabloi commented on app/product/views.py:
El endpoint debe pedir un token de authorization

pedrodepabloi reviewed (changes requested):
Falta la autorización para utilizar el endpoint

1862507J committed:
feat: Add authentication and authorization checks to ProductPUT

This commit adds authentication and authorization checks to the `ProductPUT` class. It ensures that only authenticated users can make PUT requests and only staff members can update the products. If a user is not authenticated or is not a staff member, a 403 Forbidden response is returned.

Note: The commit message has been generated based on the provided code changes and recent repository commits.
